File: NeetCode_150/Arrays_And_Hashing/Two_Sum.java
import java.util.HashMap;
import java.util.Scanner;

public class Two_Sum {

    public static int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> mpp = new HashMap<>();
        int ans[] = new int[2];
        for (int i = 0; i < nums.length; i++) {
            int req = target - nums[i];
            if (mpp.containsKey(req)) {
                ans[0] = mpp.get(req);
                ans[1] = i;
            }
            mpp.put(nums[i], i);
        }
        return ans;
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        
        in.close();
    }

}

File: NeetCode_150/Arrays_And_Hashing/Two_Sum.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
    int n;
    cout<<"Enter NUmber of elements : ";
    cin>>n;
    int arr[n];
    cout<<"Enter array elements : ";
    for(int i=0;i<n;i++){
        cin>>arr[i];
    }
    cout<<"Enter target : ";
    int trgt;
    cin>>trgt;
    int ans[2];
    unordered_map<int,int> mpp;
    for(int i=0;i<n;i++){
        int req = trgt-arr[i];
        if(mpp.find(req)!=mpp.end()){
            ans[0] = mpp[req];
            ans[1] = i;
        }
        mpp[arr[i]] = i;
    }
    cout<<"The indices are : "<<ans[0]<<" "<<ans[1]<<"\n";
} 